src/Bicep.LangServer/Handlers/BicepTextDocumentSyncHandler.cs

```diff
@@ -63,8 +76,19 @@ public override Task<Unit> Handle(DidOpenTextDocumentParams request, Cancellatio
             {
                 bicepConfigChangeHandler.HandleBicepConfigOpenEvent(documentUri);
             }
+
             //new compliation manager 
-            this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, request.TextDocument.Text, request.TextDocument.LanguageId);
+            var fileExtension = Path.GetExtension(request.TextDocument.Uri.Path);
+
+            if (fileExtension == ".bicep")
```

Comment:
For Bicep files, we should allow **any** extension other than `.bicepparam`. This is because we want to permit scenarios when the user opens a new window and sets the language to "Bicep" (at which point, the URI won't have an extension)